frontend/src/util/auth.tsx

```diff
@@ -77,9 +82,82 @@ class SimpleAuth {
 	}
 }
 
+class PasswordAuth {
```

Comment:
Implements Password auth mode.

When initialized we assign current user only if a token exists.
(Would've been nice to check the validity of the token?)

attempts an "actual" signin using the email and password
sets token to sessionStorage on sucess.

signOut just removes the token from local storage

Reply:
> (Would've been nice to check the validity of the token?)

could we have the `/login` endpoint (or another one like `/user`) return the user details when a valid token is provided? that way the user here can be populated from that, instead of duplicating the demo user details (and this will allow for a future state where we may actually have more than one user with password auth)

Reply:
I updated the ```/login``` endpoint to return user data which is used instead of duplicating demo user details